src/Sarif/Core/Notification.cs

```diff
@@ -12,14 +14,19 @@ public override string ToString()
         {
             var sb = new System.Text.StringBuilder();
 
-            sb.Append(this.Locations?[0].PhysicalLocation?.ArtifactLocation?.Uri);
-            sb.Append(" : ").Append(this.Descriptor.Id);
-            sb.Append(" : ").Append(this.AssociatedRule?.Id);
-            sb.Append(" : ").Append(this.Level);
+            if (this.TimeUtc != default) { sb.Append($"{this.TimeUtc}"); }
```

Comment:
>this [](http://example.com/codeflow?start=16&length=4)

Provides a very nice way in the debugger to see what unexpected stuff is getting persisted to configuration and tool notifications, often something like an unhandled engine exception.